cmd/maya-apiserver/cstor-operator/spc/storagepool_create.go

```diff
@@ -152,80 +152,81 @@ func poolCreateWorker(pool *apis.CasPool) error {
 	return nil
 }
 
-// withDisks builds the CasPool object by filling details like diskList,nodeName etc.
+// withBlockDevices builds the CasPool object by filling details like
+// blockDeviceList,nodeName etc.
 // Some of the fields of the CasPool object is passed to CAS engine.
 // CasPool object(type) is the contract on which CAS engine is instantiated for cStor pool creation.
 func (pc *PoolCreateConfig) withDisks(casPool *apis.CasPool, spc *apis.StoragePoolClaim) (*apis.CasPool, error) {
-	// getDiskList will hold node and disks attached to it to be used for storagepool provisioning.
-	nodeDisks, err := pc.NodeDiskSelector()
+	// getDiskList will hold node and block devices attached to it to be used for storagepool provisioning.
+	nodeBDs, err := pc.NodeBlockDeviceSelector()
 	if err != nil {
 		return nil, errors.Wrapf(err, "aborting storagepool create operation as no node qualified")
 	}
-	if len(nodeDisks.Disks.Items) == 0 {
-		return nil, errors.New("aborting storagepool create operation as no disk was found")
+	if len(nodeBDs.BlockDevices.Items) == 0 {
+		return nil, errors.New("aborting storagepool create operation as no block device was found")
 	}
 
 	// Fill the node name to the CasPool object.
-	casPool.NodeName = nodeDisks.NodeName
+	casPool.NodeName = nodeBDs.NodeName
 	//casPool.DiskList = nodeDisks.Disks.Items
 	//TODO: Improve Following Code
 	if spc.Spec.PoolSpec.PoolType == string(apis.PoolTypeStripedCPV) {
-		for _, disk := range nodeDisks.Disks.Items {
-			var diskList []apis.CspDisk
-			var group apis.DiskGroup
-			disk := apis.CspDisk{
-				Name:        disk,
+		for _, bd := range nodeBDs.BlockDevices.Items {
+			var bdList []apis.CspBlockDevice
+			var group apis.BlockDeviceGroup
+			blockDevice := apis.CspBlockDevice{
+				Name:        bd,
 				InUseByPool: true,
 			}
-			devID, err := pc.getDeviceID(disk.Name)
+			devID, err := pc.getDeviceID(blockDevice.Name)
 			if err != nil {
-				return nil, errors.Wrapf(err, "failed to get dev is for disk %s for spc %s", disk.Name, spc.Name)
+				return nil, errors.Wrapf(err, "failed to get dev Id for block device %s for spc %s", blockDevice.Name, spc.Name)
 			}
-			disk.DeviceID = devID
-			diskList = append(diskList, disk)
-			group = apis.DiskGroup{
-				Item: diskList,
+			blockDevice.DeviceID = devID
+			bdList = append(bdList, blockDevice)
+			group = apis.BlockDeviceGroup{
+				Item: bdList,
 			}
-			casPool.DiskList = append(casPool.DiskList, group)
+			casPool.BlockDeviceList = append(casPool.BlockDeviceList, group)
 		}
 		return casPool, nil
 	}
 	count := nodeselect.DefaultDiskCount[spc.Spec.PoolSpec.PoolType]
-	for i := 0; i <= len(nodeDisks.Disks.Items)/count; i = i + count {
-		var diskList []apis.CspDisk
-		var group apis.DiskGroup
+	for i := 0; i <= len(nodeBDs.BlockDevices.Items)/count; i = i + count {
```

Comment:
this logic looks to be wrong. We are checking for number of groups by dividing len(Items) with group_size (i.e., count). But, increment is happening with group size which is not correct.

Reply:
@sonasingh46 ^^

Reply:
Consider mirrored case: 
Disks : [d1, d2, d3, d4, d5]
No. of possible group = 2  [ len(disks)/2 = 2 ]
possible i values : [0,2]
Inside J Loop :
when i=0 : take disk index 0 and 1 from item
when i=2 : take from index 2 and 3.

Similarly for other cases.

Reply:
@vishnuitta

Reply:
check with 6 disks

Reply:
Yes.. It fails with 8 disks.
Nice pick. Thanks